Add advanced example to CI/CD guide (#12188)
PrefectHQ/prefect: docs/guides/ci-cd.md

You can view the logs from each workflow step as they run. The `Prefect Deploy` step will include output about your image build and push, and the creation/update of your deployment.

> ```bash

Reviewer: nit: i think we typically use the terminal div class instead of `bash` like [this](https://github.com/PrefectHQ/prefect/blob/main/docs/guides/deployment/overriding-job-variables.md?plain=1#L181)

Reviewer 2: that was my only comment for the output

└─────────────────────┴─────────┴─────────┘
```

## Advanced example

> In more complex scenarios, CI/CD processes must often make a number of additional considerations to enable a smooth development workflow:

Reviewer 3: ```suggestion
In more complex scenarios, CI/CD processes often need to accomodate several additional considerations to enable a smooth development workflow:
```
